archive/multisource.go

```diff
@@ -123,14 +120,24 @@ func (m *multiSource) chunkWalk(ctx context.Context, zctx *resolver.Context, sf
 	return Walk(ctx, m.ark, func(chunk Chunk) error {
 		so := func() (driver.ScannerCloser, error) {
 			paths := make([]string, len(m.altPaths))
+			var size int64
 			for i, input := range m.altPaths {
-				paths[i] = chunk.Localize(m.ark, input).String()
+				u := chunk.Localize(m.ark, input)
+				// XXX add size to chunk meta data so we don't need to do this
```

Comment:
I think it's a good idea to add the chunk file size to metadata, if you want to introduce that in this PR, I'd be good with that.

Reply:
Will do 👍

Reply:
I added file size to chunk meta data but am realizing we don't have chunk meta data for index types, still need this iosrc.Stat for now.